### PR TITLE
fixes mappingg loading issues

### PR DESCRIPTION
**Pull request by Bod9001**

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs [hunk 1]
```diff
@@ -119,7 +119,7 @@ protected virtual void OnEnable()
 	        {
 		        if (Material == null)
 		        {
-			        Material = Resources.GetBuiltinResource<Material>("Sprites-Default.mat");
+			        Material = Resources.GetBuiltinResource<Material>("Default Light Material.mat");
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs:122]: why the change?

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 2]
```diff
@@ -338,11 +341,17 @@ public void SetSpriteVariant(int spriteVariant, bool networked = true)
...
 	{
+		Init();
 		if (setColour == null)
 		{
 			UpdateImageColor();
 		}
 
+		if (setColour == null)
+		{
+			Loggy.LogError("AAAAAAAAAAAA");
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs]: AAAAAAAAAAAAAAAAAAAAAAAA is not an informative error log.
also follow our logging standards.

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs [hunk 3]
```diff
@@ -138,13 +137,20 @@ public static Matrix MakeNewMatrix(string Name = "Matrix")
 
 		if (Object == null)
 		{
-			Object = Spawn.ServerPrefab(MatrixPrefab).GameObject;
+			Object = Spawn.ServerPrefab(MatrixPrefab, parent: Instance.transform).GameObject;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: that doesn't sound right.
why are you parenting it to a static manager?

File: UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.SceneList.cs [hunk 4]
```diff
@@ -99,7 +99,7 @@ public IEnumerator RoundStartServerLoadSequence()
 	IEnumerator ServerLoadSpaceScene(SubsceneLoadTimer loadTimer)
 	{
 		loadTimer.IncrementLoadBar($"Loading the void of time and space");
-		yield return StartCoroutine(LoadSubScene("SpaceScene", loadTimer, default, SceneType.Space));
+		yield return StartCoroutine(LoadSubScene("AdditionalScenes/EmptySpaceYesEmptySpace.json", loadTimer, default, SceneType.Space));
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.SceneList.cs:102]: just call it empty space

File: UnityProject/Assets/Scripts/MapSaver/MapLoader.cs [hunk 5]
```diff
@@ -310,7 +310,8 @@ public static void ProcessIndividualObject(MapSaver.CompactObjectMapData Compact
 
 			if (string.IsNullOrEmpty(prefabData.SortPath) == false)
 			{
-				var Parent = Matrix.MetaTileMap.ObjectLayer.transform;
+				Transform Parent = null;
+				Parent = Matrix.MetaTileMap.ObjectLayer.transform;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/MapSaver/MapLoader.cs:314]: why?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs [hunk 6]
```diff
@@ -27,7 +27,7 @@ namespace TileManagement
 	[ExecuteInEditMode]
 	public class MetaTileMap : MonoBehaviour
 	{
-		public int TargetMSpreFrame = 5;
+		public int TargetMSpreFrame = 20;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs:30]: what does this do?

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs:122]: it also complaining about not being able to find them but this doesn't work as well

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs]: oops is debug

File: UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs:122]: maybe its the right resource type?

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: because otherwise it will try finding a matrix and it won't find one and then it will proceed error, Because this can initialise by spawning in the space matrix , this is the matrix item as well, it also gets moved in the lines Right after the

File: UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.SceneList.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.SceneList.cs:102]: well name is funny

File: UnityProject/Assets/Scripts/MapSaver/MapLoader.cs (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/MapSaver/MapLoader.cs:314]: was debuging and trying different ways so it ended being the old approach

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs (hunk 6 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs:30]: how much time is spent placing tiles,  since it takes ages for Valley To load in

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs (hunk 3 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: Parenting a game object under a static manager sounds like a hack that will come to bite us later.
I would try to find a matrix for it to attach to based on where it's located in world position, instead of putting it under something that isn't designed to hold any gameplay objects.

File: UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.SceneList.cs (hunk 4 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.SceneList.cs:102]: true, resolved.

File: UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lighting System/Light2D/CustomSprite.cs:122]: idk

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: it gets repainted instantly

File: UnityProject/Assets/Scripts/MapSaver/MapLoader.cs (hunk 5 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/MapSaver/MapLoader.cs:314]: ok, if you're going to make any changes to the code again; just put it back to its one line state.

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: reparented*

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: in that case, it's okay.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: cool

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs:140]: I would leave a comment explaining this behavior so if someone wants to grab the matrix of an object that just spawned or if a crowbar is stuck under matrix manager, they can understand why its returning matrix manager.

**Review by MaxIsJoe — APPROVED** [review]
